src/components/forms/account/AccountForm.tsx

```diff
@@ -302,64 +298,36 @@ export default function AccountForm({
 async function onSubmit(
   data: FormValues,
   action: 'add' | 'update' | 'delete',
-  router: AppRouterInstance,
   onSave: Function,
 ) {
-  // TODO: check if currency exists, if not, calculate the rate
-  // against the parent and against main currency
   const account = Account.create({
     ...data,
     guid: data.guid || undefined,
   });
-  // For some reason the beforeInsert doesn't work when updating an account.
-  // The code is not different from when we add an account and it works there...
-  account.setPath();
 
-  if (action === 'add') {
+  if (action === 'add' || action === 'update') {
     await account.save();
-    mutate(`/api/accounts/${account.guid}`);
-    mutate('/api/accounts');
-    if (data.balance) {
+    if (action === 'add' && data.balance) {
       await createBalance(data, account);
     }
-  } else if (action === 'update') {
-    await account.save();
-    mutate(`/api/accounts/${account.guid}`);
-    mutate('/api/accounts');
   } else if (action === 'delete') {
-    await Account.remove(account);
-    mutate(`/api/accounts/${account.guid}`);
-    mutate('/api/accounts');
-    router.replace('/dashboard/accounts');
```

Comment:
this replace has been moved out, in the page that renders the button because we won't always have to send the user to the home page.